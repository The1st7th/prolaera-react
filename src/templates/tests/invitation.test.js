import renderer from 'react-test-renderer';
import writeFile from '../../helpers/writeFileHelper';
import eventInvite from '../../templates/tests/json/eventInvite.json';
import invitationEmail from '../builders/eventInvitationBuilder';
import inlineCss from 'inline-css';
var request = require('request');

const _sendEmail = (template, emails) => {
  return new Promise((resolve, reject) => {
    request(
      {
        url: 'https://api-dev.prolaera.com/v1/mailer',
        method: 'PUT',
        json: {
          template,
          emails
        }
      },
      (error, response, body) => {
        if (error) {
          console.error(error);
          return reject(error);
        }
        resolve({
          response,
          body
        });
      }
    );
  });
};
describe('invitation Email', () => {
  const logoUrl = 'https://assets.prolaera.com/prolaeraLogo_fullText.png';

  it('checks snapshot', async () => {
    const emailHtml = await invitationEmail(eventInvite, logoUrl);
    const emailComponent = renderer.create(emailHtml);
    const registrationJson = emailComponent.toJSON();
    expect(registrationJson).toMatchSnapshot();
  });

  it('writes an html file', async () => {
    const email = await invitationEmail(eventInvite, logoUrl);
    const newEmail = await inlineCss(email, {
      url: ' ',
      preserveMediaQueries: true,
      applyWidthAttributes: true,
      applyTableAttributes: true
    });
    await _sendEmail(newEmail, ['aflupton@gmail.com', 'andrew@prolaera.onmicrosoft.com', 'teagan@prolaera.com']);
    const saved = await writeFile(newEmail, 'inviteTest.html');
    expect(saved).toEqual(true);
  });
});
