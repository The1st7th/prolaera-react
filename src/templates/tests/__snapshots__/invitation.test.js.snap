// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`invitation Email checks snapshot 1`] = `
"<!DOCTYPE html PUBLIC \\"-//W3C//DTD XHTML 1.0 Strict//EN\\" \\"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\\"><html lang=\\"en\\" xmlns=\\"http://www.w3.org/1999/xhtml\\"><head><meta http-equiv=\\"Content-Type\\" content=\\"text/html; charset=utf-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1.0\\"/><title>You have been Invited</title><style type=\\"text/css\\">@font-face {
	font-family: source-sans-pro;
	url(&#x27;https://fonts.googleapis.com/css?family=Source+Sans+Pro&#x27;);

}
html {
	font-family: source-sans-pro, sans-serif;
}

.textMuted {
	color: #333333;
}

#subHeaderWrapper {
	padding-top: 20px;
	padding-bottom: 0px;
}

.innerHtmlStyles {
	margin-top: 10px;
}

.innerHtmlStyles p, .innerHtmlStyles ol, .innerHtmlStyles ul  {
	margin-top: 10px;
}

button.buttonWidths {
	-webkit-appearance: none;
	-moz-appearance: none;
}

@media
only screen and (max-width: 768px)  {
	html {
		font-family: source-sans-pro, sans-serif;
	}

	body{
		background-color: #F7F7F7;
		margin-left: auto;
		margin-right: auto;
	}

	

	/* Force table to not be like tables anymore */
	table, thead, tbody, th, td, tr {
		display: block;		
	}
	
	#courseCertificate {
		max-width: 100%;
	}

	.courseCertificate {
		width: 100%;
		margin: auto;
	}
	
	.emailHeader td {
		width: 100%;
	}

	.emailHeader td img {
		margin-left: auto;
		margin-right: auto;
	} 

	.emailFooter {
		width:500px
		color: #F7F7F7;
		margin-right: auto;
		margin-left: auto;
		font-size: 10pt;
	}

	.emailFooter td {
	
		color: #F7F7F7;
		margin-right: auto;
		margin-left: auto;
		font-size: 10pt;
		
	}

	.report td {
		/* Behave  like a &quot;row&quot; */
		border: none;
		border-bottom: 1px solid #eee;
    	position: relative;
    	text-align: top;
	}

	.report td:before {
		/* Now like a table header */
		position: absolute;
		/* Top/left values mimic padding */
		top: 6px;
		left: 6px;
		width: 45%;
		padding-right: 10px;
		white-space: nowrap;
	}
	
	.customButton:hover {
		color:black;
		
	}
	#div1{
		text-align:center;
	}
	#one{
		
		position:relative;
		top:5px;
		paddingTop: 5px;
		height:23px;
		width:100px
	}
	#div2{
		text-align:center;
		padding-top:7px;
	}
}
	@media (min-width: 769px) {

		html{
			padding-left:25%;
			font-family: source-sans-pro, sans-serif;
		}
		body{
			background-color: #F7F7F7;
			margin-left: auto;
			margin-right: auto;
			max-width:55%;
			
		}
		/* Force table to not be like tables anymore */
		table, thead, tbody, th, td, tr {
			// display: block;		
		}

		#courseCertificate {
			max-width: 100%;
		}
		.courseCertificate {
			width: 100%;
			margin: auto;
		}
		
		.emailHeader td {
			
		}
	
		.emailHeader td img {
			margin-left: auto;
			margin-right: auto;
		} 
	
		.emailFooter {
			padding-top:10px;
			padding-bottom:10px;
			max-height:100px;
			width:100%;
			color: #F7F7F7;
			margin-right: auto;
			margin-left: auto;
			font-size: 10pt;
		}

		.emailFooter td {
			width:100%;
			color: #F7F7F7;
			margin-right: auto;
			margin-left: auto;
			font-size: 10pt;
			
		}

	
		.report td {
			/* Behave  like a &quot;row&quot; */
			border: none;
			border-bottom: 1px solid #eee;
			position: relative;
			text-align: top;
		}
	
		.report td:before {
			/* Now like a table header */
			position: absolute;
			/* Top/left values mimic padding */
			top: 6px;
			left: 6px;
			width: 45%;
			padding-right: 10px;
			white-space: nowrap;
		}
	
		.customButton:hover {
			color:black;
		
	  }
    #div1 {
		float:left;
		margin-left:5%;
	}
	#span1{
		float:left;
		padding:10px 5px 10px 10px;
	}
	#div2{
		text-align:right;
		padding:10px 10px 10px 10px;
		margin-left:5%
	}
	#one{
		paddingTop: 5px;
		height:23px;
		width:100px;
		position:relative;
		top:5px;
	}
	#footerId{
		padding-top:20px;
		padding-bottom:0px;
	}
	
/*End of media query*/
}</style></head><body style=\\"width:100%;margin:0;padding:0;-webkit-text-size-adjust:100%;-ms-text-size-adjust:100%\\"><table width=\\"100%\\" height=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" align=\\"left\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\" valign=\\"top\\"><table width=\\"600\\" align=\\"center\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><table class=\\"emailHeader\\" align=\\"center\\" width=\\"100%\\" style=\\"background-color:#F7F7F7\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><table align=\\"center\\" cellSpacing=\\"0\\" style=\\"padding:15px;max-width:584px;margin-top:25px;margin-bottom:20px\\"><tr align=\\"center\\"><td width=\\"292px\\"><img alt=\\"logo\\" src=\\"https://assets.prolaera.com/prolaeraLogo_fullText.png\\" width=\\"199.15254237288136\\" height=\\"47\\" style=\\"display:block;outline:none;border:none;text-decoration:none\\"/></td></tr></table></td></tr></tbody></table><div id=\\"subHeaderWrapper\\" style=\\"background-color:#FFFFFF;margin-left:auto;margin-right:auto;font-style:normal;color:black;font-weight:500;display:block;letter-spacing:1pt\\"><table class=\\"subHeader\\" align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td><table id=\\"subHeaderTable\\" style=\\"text-align:center;margin-left:auto;margin-right:auto\\"><tr><td><p style=\\"font-size:18pt;margin-bottom:5px\\">You have been</p></td></tr><tr><td><p style=\\"font-size:28px;margin-top:0px;margin-bottom:0px\\">Invited</p></td></tr><tr style=\\"margin:auto;padding:0px 10px 10px 10px\\"><td style=\\"width:100%\\" align=\\"center\\"><div style=\\"max-width:85%;letter-spacing:1pt;padding:20px 20px 0px 20px\\">You have been invited to a new event! More details are included below. Please click the register button below if you are able to attend.</div><div style=\\"max-width:85%\\"><div></div></div></td></tr></table></td></tr></tbody></table></div><div id=\\"subHeaderWrapper\\" style=\\"display:block;margin:auto;text-align:center;font-style:normal;font-weight:400;background-color:#FFFFFF\\"><table align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\"><div style=\\"width:90%;margin:auto\\"><div><table class=\\"courseReview\\" align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"center\\" style=\\"padding:20px\\"><div style=\\"width:100%;height:20px;border-bottom:1px solid grey;text-align:center;box-sizing:border-box;position:relative;margin:auto;padding:10px\\"><span style=\\"font-size:14pt;background-color:#FFFFFF;padding:0px 20px;box-sizing:border-box;margin:0\\">Event Information</span></div></td></tr></tbody></table></div></div><h4 align=\\"center\\"><a href=\\"https://app.prolaera.com/#/events/e2b650ed-6ee4-4f5b-a614-f72562790ebe\\" style=\\"text-decoration:underline;color:#2F4050;height:100%\\"><em>Fraud in the Digital Age (Group-Live)</em></a></h4></td></tr><tr><td align=\\"left\\"><div style=\\"padding-left:40px\\"><h4>Start Time:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\">June 14, 2018</div><h4>End Time:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\">June 15, 2018</div><h4 style=\\"margin-bottom:0px\\">Recommended CPE Credit(s):</h4><ul style=\\"margin-top:10px\\"><li style=\\"padding:5px\\">Auditing - 12 Hour(s)</li><li style=\\"padding:5px\\">Uncategorized - 4 Hour(s)</li><li style=\\"padding:5px\\">Other - 0.1 Hour(s)</li></ul><h4>Location:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\"><a style=\\"text-decoration:underline;color:inherit\\" href=\\"https://www.google.com/maps/place/The White House, 1600 Pennsylvania Avenue Northwest, Washington, DC, USA\\"><span style=\\"font-weight:normal\\">The White House, 1600 Pennsylvania Avenue Northwest, Washington, DC, USA</span></a></div><h4>Delivery Method:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\">Group-Live</div><h4>Price: </h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\">$200000.00</div><h4>Target Audience:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\">CPAs who perform audits, some other guys, literally anyone with a brain and 200,000 dollars.</div><h4 style=\\"margin-bottom:0px\\">Prep:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\"><div><u>None needed.&nbsp;</u></div></div><h4 style=\\"margin-bottom:0px\\">Prerequisites:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\"><ul><li>Familiarity with fraud detection methods</li><li>Deep-seated hatred for fraud</li><li>Love of avoiding fraud</li><li>Two hundred thousand dollars</li></ul></div><h4>Level:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\">Advanced</div><h4 style=\\"margin-bottom:0px\\">Learning Objectives:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:norma;\\"><ul><li>Objective 1</li><li>Objective 2</li><li>Objective 3</li><li>Objective 3 + 1</li></ul></div><h4 style=\\"margin-bottom:0px\\">Summary:</h4><div class=\\"innerHtmlStyles\\" style=\\"font-weight:normal\\"><p><b>This course is about</b>&nbsp;detecting fraud. On top of that, this course is also about many other things. Here is a list of some of the things that will be covered in this course.&nbsp;</p><ul><li>Stuff</li><li>Things</li><li>Topics</li></ul><p>Please bring a pen and paper, and laptop. If you don't have any of those, tough luck.</p><div><span style=\\"background-color: rgb(85, 98, 113); color: rgb(123, 136, 152); font-family: &quot;Mercury SSm A&quot;, &quot;Mercury SSm B&quot;, Georgia, Times, &quot;Times New Roman&quot;, &quot;Microsoft YaHei New&quot;, &quot;Microsoft Yahei&quot;, 微软雅黑, 宋体, SimSun, STXihei, 华文细黑, serif; font-size: 1.1875em;\\"><br></span></div></div></div></td></tr><tr><td align=\\"center\\" style=\\"width:100%;padding-bottom:20px\\"><div style=\\"display:inline-block;margin:auto;padding:20px\\"><div class=\\"customButton\\" align=\\"center\\" style=\\"text-align:center;height:100%;width:100%\\"><a href=\\"https://app.prolaera.com/#/events/e2b650ed-6ee4-4f5b-a614-f72562790ebe\\"><button class=\\"buttonWidths\\" style=\\"height:100%;padding:14px 14px 14px 14px;color:#FFFFFF;font-size:10pt;width:100%;text-decoration:none;border-radius:3px;border-style:none;border-color:#72C02C;background-color:#72C02C\\">Register</button></a></div></div></td></tr></tbody></table></div><div id=\\"subHeaderWrapper\\" style=\\"margin-left:auto;margin-right:auto;margin-bottom:0px;font-family:source-sans-pro, sans-serif;padding:20px;background-color:white\\"><table class=\\"subFooter\\" align=\\"center\\" width=\\"100%\\" cellPadding=\\"0\\" cellSpacing=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td align=\\"justify\\" style=\\"padding-left:30px;padding-right:30px\\"><p>All you need to do is RSVP. If you decide to go, you will be navigated to the Prolaera website. Don&#x27;t have an account with us? No problem, it only takes a few seconds to get one started!</p><p></p></td></tr></tbody></table></div><div><table id=\\"footerId\\" class=\\"emailFooter\\" align=\\"center\\" cellSpacing=\\"20\\" style=\\"background-color:#333333;padding-top:5px\\" height=\\"60\\" cellPadding=\\"0\\" border=\\"0\\" valign=\\"top\\"><tbody><tr><td><div id=\\"div1\\"><span id=\\"span1\\" style=\\"color:#F7F7F7;font-size:10pt;font-weight:100;letter-spacing:1pt\\">Powered By </span><img id=\\"one\\" src=\\"https://gallery.mailchimp.com/eb930e30a15249a18d658c183/images/196fa069-3113-464c-a304-a1aa9f99fe22.png\\" alt=\\"react\\"/></div><div id=\\"div2\\" style=\\"color:#F7F7F7;font-size:10pt;font-weight:100;letter-spacing:1pt\\">© Prolaera 2018</div></td></tr></tbody></table></div></tbody></table></td></tr></tbody></table></body></html>"
`;
